src/vs/workbench/browser/parts/titlebar/titlebarPart.ts

```diff
@@ -128,6 +135,39 @@ export class BrowserTitleService extends MultiWindowParts<BrowserTitlebarPart> i
 		}));
 	}
 
+	private registerCommands(): void {
+		this._register(CommandsRegistry.registerCommand({
+			id: 'registerWindowTitleVariable',
+			handler: (accessor: ServicesAccessor, name: string, contextKey: string) => {
+				if (accessor.get(IContextKeyService).getContextKeyValue(contextKey) === undefined) {
+					console.error(`Unable to register window title variable: Context key '${contextKey}' is not defined.`);
+					return;
+				}
+				this.registerVariables([{ name, contextKey }]);
+			},
+			metadata: {
+				description: 'Registers a new title variable',
+				args: [
+					{ name: 'name', schema: { type: 'string' }, description: 'The name of the variable to register' },
+					{ name: 'contextKey', schema: { type: 'string' }, description: 'The context key to use for the value of the variable' }
+				]
+			}
+		}));
+
+		this._register(CommandsRegistry.registerCommand({
+			id: 'unregisterWindowTitleVariable',
+			handler: (_, name: string) => {
+				this.unregisterVariables([name]);
+			},
+			metadata: {
+				description: 'Unregister a title variable',
+				args: [
+					{ name: 'name', schema: { type: 'string' }, description: 'The name of the variable to unregister' },
+				]
+			}
+		}));
```

Comment:
I would not do this, why would we allow extensions to unregister any kind of variables they did not create?